hpx/util/coroutine/coroutine.hpp

```diff
@@ -340,40 +288,8 @@ namespace hpx { namespace util { namespace coroutines
       return !empty() && !exited() && !waiting();
     }
 
-#if defined(HPX_HAVE_GENERIC_CONTEXT_COROUTINES)
```

Comment:
We don't. As the ticket said (see #1584)

> the pp constant  HPX_HAVE_GENERIC_CONTEXT_COROUTINES  is currently used for two different things

This patch removes one of the use of `HPX_HAVE_GENERIC_CONTEXT_COROUTINES`